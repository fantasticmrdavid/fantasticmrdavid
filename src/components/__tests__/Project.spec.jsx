import React from 'react';
import { shallow } from 'enzyme';
import Ui from 'components/Project/ui';
import projects from 'data/projects';

const props = {
  current: undefined,
  isCurrent: false,
};

projects.map((p) => {
  describe(`Component -> Project -> ${p.title}`, () => {
    it('should render the Project UI without throwing an error', () => {
      expect(shallow(<Ui {...props} {...p} />)).toMatchSnapshot();
    });
  });

  return p;
});
