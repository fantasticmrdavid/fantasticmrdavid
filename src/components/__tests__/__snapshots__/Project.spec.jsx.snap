// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Project -> General Assembly should render the Project UI (current) without throwing an error 1`] = `
<styles__Container
  current={true}
  otherCurrent={false}
>
  <styles__Tile
    current={true}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="ga"
  >
    <styles__Image
      current={true}
      srcOff="/assets/images/projects/ga_off.jpg"
      srcOn="/assets/images/projects/ga_on.jpg"
      target="ga"
    />
    <styles__TileContent
      current={true}
    >
      <styles__Title>
        General Assembly
      </styles__Title>
      <styles__Tagline>
        Lead Instructor
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={true}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="Starting out as a co-working space, General Assembly has grown into a pioneer in education and career transformation, specializing in today’s most in-demand skills. They are a leading source for training, staffing, and career transitions, fostering a flourishing community of professionals pursuing careers they love.<br /><br />I taught the Front End Web Development (FEWD) part time course at General Assembly's Melbourne campus. Throughout the 10 week course, students learnt HTML, CSS, JavaScript, and fundamental programming skills to build their own website project. It was immensely rewarding with students moving on to roles ranging from Junior Developer to Project Manager."
    images={
      Object {
        "tileOff": "/assets/images/projects/ga_off.jpg",
        "tileOn": "/assets/images/projects/ga_on.jpg",
      }
    }
    isCurrent={true}
    parentLoading={false}
    tagline="Lead Instructor"
    target="ga"
    technologies="HTML, CSS, JavaScript, PHP"
    title="General Assembly"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
          Array [
            "legoFlinders",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    url="https://generalassemb.ly"
  />
</styles__Container>
`;

exports[`Component -> Project -> General Assembly should render the Project UI (not current) without throwing an error 1`] = `
<styles__Container
  current={false}
  otherCurrent={false}
>
  <styles__Tile
    current={false}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="ga"
  >
    <styles__Image
      current={false}
      srcOff="/assets/images/projects/ga_off.jpg"
      srcOn="/assets/images/projects/ga_on.jpg"
      target="ga"
    />
    <styles__TileContent
      current={false}
    >
      <styles__Title>
        General Assembly
      </styles__Title>
      <styles__Tagline>
        Lead Instructor
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={false}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="Starting out as a co-working space, General Assembly has grown into a pioneer in education and career transformation, specializing in today’s most in-demand skills. They are a leading source for training, staffing, and career transitions, fostering a flourishing community of professionals pursuing careers they love.<br /><br />I taught the Front End Web Development (FEWD) part time course at General Assembly's Melbourne campus. Throughout the 10 week course, students learnt HTML, CSS, JavaScript, and fundamental programming skills to build their own website project. It was immensely rewarding with students moving on to roles ranging from Junior Developer to Project Manager."
    images={
      Object {
        "tileOff": "/assets/images/projects/ga_off.jpg",
        "tileOn": "/assets/images/projects/ga_on.jpg",
      }
    }
    isCurrent={false}
    parentLoading={false}
    tagline="Lead Instructor"
    target="ga"
    technologies="HTML, CSS, JavaScript, PHP"
    title="General Assembly"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
          Array [
            "legoFlinders",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    url="https://generalassemb.ly"
  />
</styles__Container>
`;

exports[`Component -> Project -> KindyNow should render the Project UI (current) without throwing an error 1`] = `
<styles__Container
  current={true}
  otherCurrent={false}
>
  <styles__Tile
    current={true}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="kindynow"
  >
    <styles__Image
      current={true}
      srcOff="/assets/images/projects/kindynow_off.jpg"
      srcOn="/assets/images/projects/kindynow_on.jpg"
      target="kindynow"
    />
    <styles__TileContent
      current={true}
    >
      <styles__Title>
        KindyNow
      </styles__Title>
      <styles__Tagline>
        Lead Front-End Engineer
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={true}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="KindyNow provides Australia's leading app and web service for helping parents find daycare for their children. As Lead Front-End engineer I architected and developed the company's flagship web assets, including a new child care search platform (Childcare Near Me), a rebuild of their existing booking management app (KindyNow), and admin tools for internal management of centres and bookings.<br /><br />I also managed and trained junior engineers, collaborated with the marketing team to prioritise and build features to maximise conversion goals, and provided guidance on UX and product strategy."
    images={
      Object {
        "tileOff": "/assets/images/projects/kindynow_off.jpg",
        "tileOn": "/assets/images/projects/kindynow_on.jpg",
      }
    }
    isCurrent={true}
    media={
      Array [
        Object {
          "platform": "mobile",
          "slug": "cnm_mobile_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_mobile_preview.jpg",
          "title": "Childcare Near Me mobile site",
          "type": "website",
          "url": "https://childcarenearme.com.au",
        },
        Object {
          "platform": "desktop",
          "slug": "cnm_search_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_search_preview.jpg",
          "title": "Childcare Near Me search",
          "type": "website",
          "url": "https://childcarenearme.com.au",
        },
        Object {
          "platform": "mobile",
          "slug": "kn_app_preview",
          "thumbnail": "/assets/images/projects/kindynow/kn_app_preview.jpg",
          "title": "KindyNow Mobile App",
          "type": "website",
          "url": "https://www.kindynow.com",
        },
        Object {
          "platform": "desktop",
          "slug": "cnm_desktop_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_desktop_preview.jpg",
          "title": "Childcare Near Me desktop site",
          "type": "website",
          "url": "https://childcarenearme.com",
        },
        Object {
          "platform": "desktop",
          "slug": "kn_desktop_preview",
          "thumbnail": "/assets/images/projects/kindynow/kn_desktop_preview.jpg",
          "title": "KindyNow desktop site",
          "type": "website",
          "url": "https://www.kindynow.com",
        },
      ]
    }
    parentLoading={false}
    products="Child care search platform, Centre management platform, Mobile app, Admin platform"
    tagline="Lead Front-End Engineer"
    target="kindynow"
    technologies="HTML, CSS, ES6, SSR React, React-Native, Redux, TypeScript, Node.js, Jest, Enzyme, AWS, CircleCI"
    title="KindyNow"
    updateCurrent={[MockFunction]}
    url="https://www.kindynow.com"
  />
</styles__Container>
`;

exports[`Component -> Project -> KindyNow should render the Project UI (not current) without throwing an error 1`] = `
<styles__Container
  current={false}
  otherCurrent={false}
>
  <styles__Tile
    current={false}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="kindynow"
  >
    <styles__Image
      current={false}
      srcOff="/assets/images/projects/kindynow_off.jpg"
      srcOn="/assets/images/projects/kindynow_on.jpg"
      target="kindynow"
    />
    <styles__TileContent
      current={false}
    >
      <styles__Title>
        KindyNow
      </styles__Title>
      <styles__Tagline>
        Lead Front-End Engineer
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={false}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="KindyNow provides Australia's leading app and web service for helping parents find daycare for their children. As Lead Front-End engineer I architected and developed the company's flagship web assets, including a new child care search platform (Childcare Near Me), a rebuild of their existing booking management app (KindyNow), and admin tools for internal management of centres and bookings.<br /><br />I also managed and trained junior engineers, collaborated with the marketing team to prioritise and build features to maximise conversion goals, and provided guidance on UX and product strategy."
    images={
      Object {
        "tileOff": "/assets/images/projects/kindynow_off.jpg",
        "tileOn": "/assets/images/projects/kindynow_on.jpg",
      }
    }
    isCurrent={false}
    media={
      Array [
        Object {
          "platform": "mobile",
          "slug": "cnm_mobile_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_mobile_preview.jpg",
          "title": "Childcare Near Me mobile site",
          "type": "website",
          "url": "https://childcarenearme.com.au",
        },
        Object {
          "platform": "desktop",
          "slug": "cnm_search_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_search_preview.jpg",
          "title": "Childcare Near Me search",
          "type": "website",
          "url": "https://childcarenearme.com.au",
        },
        Object {
          "platform": "mobile",
          "slug": "kn_app_preview",
          "thumbnail": "/assets/images/projects/kindynow/kn_app_preview.jpg",
          "title": "KindyNow Mobile App",
          "type": "website",
          "url": "https://www.kindynow.com",
        },
        Object {
          "platform": "desktop",
          "slug": "cnm_desktop_preview",
          "thumbnail": "/assets/images/projects/kindynow/cnm_desktop_preview.jpg",
          "title": "Childcare Near Me desktop site",
          "type": "website",
          "url": "https://childcarenearme.com",
        },
        Object {
          "platform": "desktop",
          "slug": "kn_desktop_preview",
          "thumbnail": "/assets/images/projects/kindynow/kn_desktop_preview.jpg",
          "title": "KindyNow desktop site",
          "type": "website",
          "url": "https://www.kindynow.com",
        },
      ]
    }
    parentLoading={false}
    products="Child care search platform, Centre management platform, Mobile app, Admin platform"
    tagline="Lead Front-End Engineer"
    target="kindynow"
    technologies="HTML, CSS, ES6, SSR React, React-Native, Redux, TypeScript, Node.js, Jest, Enzyme, AWS, CircleCI"
    title="KindyNow"
    updateCurrent={[MockFunction]}
    url="https://www.kindynow.com"
  />
</styles__Container>
`;

exports[`Component -> Project -> LEGO Flinders St should render the Project UI (current) without throwing an error 1`] = `
<styles__Container
  current={true}
  otherCurrent={false}
>
  <styles__Tile
    current={true}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="legoFlinders"
  >
    <styles__Image
      current={true}
      srcOff="/assets/images/projects/legoFlinders_off.jpg"
      srcOn="/assets/images/projects/legoFlinders_on.jpg"
      target="legoFlinders"
    />
    <styles__TileContent
      current={true}
    >
      <styles__Title>
        LEGO Flinders St
      </styles__Title>
      <styles__Tagline>
        CAD Art Project
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={true}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="The combination of visiting <a href=\\"http://www.brickventures.org.au/brickvention\\" target=\\"_blank\\">Brickvention</a>, unfulfilled childhood LEGO dreams, and a love for the great city that is Melbourne left a nagging itch that my creative brain couldn't ignore. The LDD model took a year to complete, over 43,000 parts, and many visits to the actual location. It was then converted and rendered in POV-Ray before adding final Melbournian touch ups in Photoshop."
    images={
      Object {
        "tileOff": "/assets/images/projects/legoFlinders_off.jpg",
        "tileOn": "/assets/images/projects/legoFlinders_on.jpg",
      }
    }
    isCurrent={true}
    media={
      Array [
        Object {
          "orientation": "landscape",
          "slug": "legoFlinders",
          "thumbnail": "/assets/images/projects/legoFlinders/legoFlinders_preview.jpg",
          "title": "LEGO Flinders St art print",
          "type": "image",
          "url": "/assets/images/projects/legoFlinders/legoFlinders_full.jpg",
        },
      ]
    }
    parentLoading={false}
    products="Artwork, prints"
    tagline="CAD Art Project"
    target="legoFlinders"
    technologies="LEGO Digital Designer, Photoshop, POV-Ray"
    title="LEGO Flinders St"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</styles__Container>
`;

exports[`Component -> Project -> LEGO Flinders St should render the Project UI (not current) without throwing an error 1`] = `
<styles__Container
  current={false}
  otherCurrent={false}
>
  <styles__Tile
    current={false}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="legoFlinders"
  >
    <styles__Image
      current={false}
      srcOff="/assets/images/projects/legoFlinders_off.jpg"
      srcOn="/assets/images/projects/legoFlinders_on.jpg"
      target="legoFlinders"
    />
    <styles__TileContent
      current={false}
    >
      <styles__Title>
        LEGO Flinders St
      </styles__Title>
      <styles__Tagline>
        CAD Art Project
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={false}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="The combination of visiting <a href=\\"http://www.brickventures.org.au/brickvention\\" target=\\"_blank\\">Brickvention</a>, unfulfilled childhood LEGO dreams, and a love for the great city that is Melbourne left a nagging itch that my creative brain couldn't ignore. The LDD model took a year to complete, over 43,000 parts, and many visits to the actual location. It was then converted and rendered in POV-Ray before adding final Melbournian touch ups in Photoshop."
    images={
      Object {
        "tileOff": "/assets/images/projects/legoFlinders_off.jpg",
        "tileOn": "/assets/images/projects/legoFlinders_on.jpg",
      }
    }
    isCurrent={false}
    media={
      Array [
        Object {
          "orientation": "landscape",
          "slug": "legoFlinders",
          "thumbnail": "/assets/images/projects/legoFlinders/legoFlinders_preview.jpg",
          "title": "LEGO Flinders St art print",
          "type": "image",
          "url": "/assets/images/projects/legoFlinders/legoFlinders_full.jpg",
        },
      ]
    }
    parentLoading={false}
    products="Artwork, prints"
    tagline="CAD Art Project"
    target="legoFlinders"
    technologies="LEGO Digital Designer, Photoshop, POV-Ray"
    title="LEGO Flinders St"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</styles__Container>
`;

exports[`Component -> Project -> SitePoint should render the Project UI (current) without throwing an error 1`] = `
<styles__Container
  current={true}
  otherCurrent={false}
>
  <styles__Tile
    current={true}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="sitepoint"
  >
    <styles__Image
      current={true}
      srcOff="/assets/images/projects/sitepoint_off.jpg"
      srcOn="/assets/images/projects/sitepoint_on.jpg"
      target="sitepoint"
    />
    <styles__TileContent
      current={true}
    >
      <styles__Title>
        SitePoint
      </styles__Title>
      <styles__Tagline>
        Front-End Engineer
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={true}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="SitePoint is a hub built by developers for developers with the goal of sharing knowledge and educating others on incredible Internet things. Spanning articles, courses, books, screencasts, and a huge forum community, they are a leading source for the latest news on all things web development.<br /><br />As Front-End engineer I worked within an Agile development team to maintain and upgrade the company Articles WordPress and Premium RoR platforms. The revised articles site featured a new homepage and template designs with server-side rendered front-end components. We completely renovated the Premium members platform library design as well as shifted front-end components from Rails to React and Redux."
    images={
      Object {
        "tileOff": "/assets/images/projects/sitepoint_off.jpg",
        "tileOn": "/assets/images/projects/sitepoint_on.jpg",
      }
    }
    isCurrent={true}
    media={
      Array [
        Object {
          "platform": "mobile",
          "slug": "sitepoint_mobile_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_mobile_preview.jpg",
          "title": "SitePoint mobile site",
          "type": "website",
          "url": "https://www.sitepoint.com",
        },
        Object {
          "platform": "desktop",
          "slug": "sitepoint_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_preview.jpg",
          "title": "SitePoint desktop site",
          "type": "website",
          "url": "https://www.sitepoint.com",
        },
        Object {
          "platform": "desktop",
          "slug": "sitepoint_premium_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_premium_preview.jpg",
          "title": "SitePoint Premium desktop site",
          "type": "website",
          "url": "https://www.sitepoint.com/premium",
        },
        Object {
          "platform": "mobile",
          "slug": "sitepoint_premium_mobile_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_premium_mobile_preview.jpg",
          "title": "SitePoint Premium mobile site",
          "type": "website",
          "url": "https://www.sitepoint.com/premium",
        },
      ]
    }
    parentLoading={false}
    products="Company media website, Premium platform, forums"
    tagline="Front-End Engineer"
    target="sitepoint"
    technologies="HTML, CSS, ES6, React, Redux, Node.js, MobX, RoR, PHP"
    title="SitePoint"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    url="https://www.sitepoint.com"
  />
</styles__Container>
`;

exports[`Component -> Project -> SitePoint should render the Project UI (not current) without throwing an error 1`] = `
<styles__Container
  current={false}
  otherCurrent={false}
>
  <styles__Tile
    current={false}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="sitepoint"
  >
    <styles__Image
      current={false}
      srcOff="/assets/images/projects/sitepoint_off.jpg"
      srcOn="/assets/images/projects/sitepoint_on.jpg"
      target="sitepoint"
    />
    <styles__TileContent
      current={false}
    >
      <styles__Title>
        SitePoint
      </styles__Title>
      <styles__Tagline>
        Front-End Engineer
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={false}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="SitePoint is a hub built by developers for developers with the goal of sharing knowledge and educating others on incredible Internet things. Spanning articles, courses, books, screencasts, and a huge forum community, they are a leading source for the latest news on all things web development.<br /><br />As Front-End engineer I worked within an Agile development team to maintain and upgrade the company Articles WordPress and Premium RoR platforms. The revised articles site featured a new homepage and template designs with server-side rendered front-end components. We completely renovated the Premium members platform library design as well as shifted front-end components from Rails to React and Redux."
    images={
      Object {
        "tileOff": "/assets/images/projects/sitepoint_off.jpg",
        "tileOn": "/assets/images/projects/sitepoint_on.jpg",
      }
    }
    isCurrent={false}
    media={
      Array [
        Object {
          "platform": "mobile",
          "slug": "sitepoint_mobile_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_mobile_preview.jpg",
          "title": "SitePoint mobile site",
          "type": "website",
          "url": "https://www.sitepoint.com",
        },
        Object {
          "platform": "desktop",
          "slug": "sitepoint_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_preview.jpg",
          "title": "SitePoint desktop site",
          "type": "website",
          "url": "https://www.sitepoint.com",
        },
        Object {
          "platform": "desktop",
          "slug": "sitepoint_premium_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_premium_preview.jpg",
          "title": "SitePoint Premium desktop site",
          "type": "website",
          "url": "https://www.sitepoint.com/premium",
        },
        Object {
          "platform": "mobile",
          "slug": "sitepoint_premium_mobile_preview",
          "thumbnail": "/assets/images/projects/sitepoint/sitepoint_premium_mobile_preview.jpg",
          "title": "SitePoint Premium mobile site",
          "type": "website",
          "url": "https://www.sitepoint.com/premium",
        },
      ]
    }
    parentLoading={false}
    products="Company media website, Premium platform, forums"
    tagline="Front-End Engineer"
    target="sitepoint"
    technologies="HTML, CSS, ES6, React, Redux, Node.js, MobX, RoR, PHP"
    title="SitePoint"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    url="https://www.sitepoint.com"
  />
</styles__Container>
`;

exports[`Component -> Project -> Tramsformation should render the Project UI (current) without throwing an error 1`] = `
<styles__Container
  current={true}
  otherCurrent={false}
>
  <styles__Tile
    current={true}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="tramsformation"
  >
    <styles__Image
      current={true}
      srcOff="/assets/images/projects/tramsformation_off.jpg"
      srcOn="/assets/images/projects/tramsformation_on.jpg"
      target="tramsformation"
    />
    <styles__TileContent
      current={true}
    >
      <styles__Title>
        Tramsformation
      </styles__Title>
      <styles__Tagline>
        3D Printing & Design
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={true}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="A personal project that started with a failed art competition entry. Featuring a website with shop, 3D models, T-shirts, prints and a 3D printed transformable figure. I have so much respect for the engineering in those old 80's toys."
    images={
      Object {
        "tileOff": "/assets/images/projects/tramsformation_off.jpg",
        "tileOn": "/assets/images/projects/tramsformation_on.jpg",
      }
    }
    isCurrent={true}
    media={
      Array [
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_print",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_print_preview.jpg",
          "title": "Tramsformer art print",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_print_full.jpg",
        },
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_transform",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_transform_preview.jpg",
          "title": "Tramsformer transformation",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_transform_full.jpg",
        },
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_model",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_model_preview.jpg",
          "title": "Tramsformer model",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_model_full.jpg",
        },
        Object {
          "orientation": "landscape",
          "slug": "tramsformation_model_altmode",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_model_altmode_preview.jpg",
          "title": "Tramsformer model tram mode",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_model_altmode_full.jpg",
        },
        Object {
          "platform": "mobile",
          "slug": "tramsformation_mobile",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_mobile_preview.jpg",
          "title": "Tramsformation mobile site",
          "type": "website",
        },
        Object {
          "platform": "desktop",
          "slug": "tramsformation_desktop",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_desktop_preview.jpg",
          "title": "Tramsformation desktop site",
          "type": "website",
        },
      ]
    }
    parentLoading={false}
    products="3D Models, artwork, website, transformable toy"
    tagline="3D Printing & Design"
    target="tramsformation"
    technologies="HTML, CSS, jQuery, PHP, MySQL, Photoshop, 3DS Max, 3D printing"
    title="Tramsformation"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
          Array [
            "legoFlinders",
          ],
          Array [
            "ga",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</styles__Container>
`;

exports[`Component -> Project -> Tramsformation should render the Project UI (not current) without throwing an error 1`] = `
<styles__Container
  current={false}
  otherCurrent={false}
>
  <styles__Tile
    current={false}
    onClick={[Function]}
    otherCurrent={false}
    parentLoading={false}
    target="tramsformation"
  >
    <styles__Image
      current={false}
      srcOff="/assets/images/projects/tramsformation_off.jpg"
      srcOn="/assets/images/projects/tramsformation_on.jpg"
      target="tramsformation"
    />
    <styles__TileContent
      current={false}
    >
      <styles__Title>
        Tramsformation
      </styles__Title>
      <styles__Tagline>
        3D Printing & Design
      </styles__Tagline>
    </styles__TileContent>
    <styles__Close
      current={false}
    >
      X
    </styles__Close>
  </styles__Tile>
  <Memo()
    description="A personal project that started with a failed art competition entry. Featuring a website with shop, 3D models, T-shirts, prints and a 3D printed transformable figure. I have so much respect for the engineering in those old 80's toys."
    images={
      Object {
        "tileOff": "/assets/images/projects/tramsformation_off.jpg",
        "tileOn": "/assets/images/projects/tramsformation_on.jpg",
      }
    }
    isCurrent={false}
    media={
      Array [
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_print",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_print_preview.jpg",
          "title": "Tramsformer art print",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_print_full.jpg",
        },
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_transform",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_transform_preview.jpg",
          "title": "Tramsformer transformation",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_transform_full.jpg",
        },
        Object {
          "orientation": "portrait",
          "slug": "tramsformation_model",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_model_preview.jpg",
          "title": "Tramsformer model",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_model_full.jpg",
        },
        Object {
          "orientation": "landscape",
          "slug": "tramsformation_model_altmode",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_model_altmode_preview.jpg",
          "title": "Tramsformer model tram mode",
          "type": "image",
          "url": "/assets/images/projects/tramsformation/tramsformation_model_altmode_full.jpg",
        },
        Object {
          "platform": "mobile",
          "slug": "tramsformation_mobile",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_mobile_preview.jpg",
          "title": "Tramsformation mobile site",
          "type": "website",
        },
        Object {
          "platform": "desktop",
          "slug": "tramsformation_desktop",
          "thumbnail": "/assets/images/projects/tramsformation/tramsformation_desktop_preview.jpg",
          "title": "Tramsformation desktop site",
          "type": "website",
        },
      ]
    }
    parentLoading={false}
    products="3D Models, artwork, website, transformable toy"
    tagline="3D Printing & Design"
    target="tramsformation"
    technologies="HTML, CSS, jQuery, PHP, MySQL, Photoshop, 3DS Max, 3D printing"
    title="Tramsformation"
    updateCurrent={
      [MockFunction] {
        "calls": Array [
          Array [
            "kindynow",
          ],
          Array [
            "sitepoint",
          ],
          Array [
            "legoFlinders",
          ],
          Array [
            "ga",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</styles__Container>
`;
